packages/scripts/src/upgrade-tests/voting2/3_SimulateVoting.ts

```diff
@@ -253,12 +253,51 @@ async function main() {
   if (!(await isVotingV2Instance(votingV2Address))) throw new Error("Oracle is not VotingV2 instance!");
 
   const votingV2 = await getContractInstance<VotingV2Ethers>("VotingV2", votingV2Address);
+  const governorV2Address = await votingV2.owner();
 
   const emissionRate = await votingV2.emissionRate();
   const gat = await votingV2.gat();
   const unstakeCoolDown = await votingV2.unstakeCoolDown();
   const finalFee = (await store.computeFinalFee(votingToken.address)).rawValue;
 
+  console.log(" 1. Unstake from all preexisting voters...");
+  await hre.network.provider.request({ method: "hardhat_impersonateAccount", params: [governorV2Address] });
+  await hre.network.provider.send("hardhat_setBalance", [
+    governorV2Address,
+    hre.ethers.utils.parseEther("10.0").toHexString(),
+  ]);
+  const govSigner = (await hre.ethers.getSigner(governorV2Address)) as Signer;
+
+  const initialCooldDownPeriod = await votingV2.unstakeCoolDown();
+
+  // Set the cool down period to 0 so that we can unstake immediately.
+  await (await votingV2.connect(govSigner).setUnstakeCoolDown(0)).wait();
+
+  // Get all Staked events from the VotingV2 contract
+  const stakedEvents = await votingV2.queryFilter(votingV2.filters.Staked(null, null));
+
+  const uniqueAddresses = new Set<string>();
+  stakedEvents.forEach((event) => {
+    uniqueAddresses.add(event.args?.voter);
+  });
+
+  for (const address of uniqueAddresses) {
+    await hre.network.provider.request({ method: "hardhat_impersonateAccount", params: [address] });
+    const signer = (await hre.ethers.getSigner(address)) as Signer;
+    await hre.network.provider.send("hardhat_setBalance", [address, hre.ethers.utils.parseEther("10.0").toHexString()]);
+    const stake = await votingV2.callStatic.getVoterStakePostUpdate(address);
+    if (stake.isZero()) continue;
+    await (
+      await votingV2.connect(signer).requestUnstake(await votingV2.callStatic.getVoterStakePostUpdate(address))
```

Comment:
do we need to worry about not being in active reveal phase?

Reply:
also might want to execute all pending stakes as otherwise `requestUnstake` would revert